modules/cdc-ext/src/main/java/org/apache/ignite/cdc/kafka/KafkaToIgniteClientCdcStreamer.java

```diff
@@ -0,0 +1,93 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.cdc.kafka;
+
+import java.util.Collection;
+import java.util.Properties;
+import java.util.UUID;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.cdc.AbstractCdcEventsApplier;
+import org.apache.ignite.cdc.thin.CdcEventsIgniteClientApplier;
+import org.apache.ignite.client.IgniteClient;
+import org.apache.ignite.configuration.ClientConfiguration;
+import org.apache.ignite.internal.binary.BinaryContext;
+import org.apache.ignite.internal.binary.GridBinaryMarshaller;
+import org.apache.ignite.internal.client.thin.ClientBinary;
+import org.apache.ignite.internal.util.typedef.internal.A;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+/**
+ * Implementation of {@link AbstractKafkaToIgniteCdcStreamer} that streams all data changes to a destination cluster
+ * through Ignite thin client.
+ *
+ * @see IgniteClient
+ */
+public class KafkaToIgniteClientCdcStreamer extends AbstractKafkaToIgniteCdcStreamer {
+    /** Ignite thin client configuration. */
+    private final ClientConfiguration clientCfg;
+
+    /** Ignite thin client. */
+    private IgniteClient client;
+
+    /**
+     * @param clientCfg Ignite thin client configuration.
+     * @param kafkaProps Kafka properties.
+     * @param streamerCfg Streamer configuration.
+     */
+    public KafkaToIgniteClientCdcStreamer(ClientConfiguration clientCfg, Properties kafkaProps,
+        KafkaToIgniteCdcStreamerConfiguration streamerCfg) {
+        super(kafkaProps, streamerCfg);
+
+        A.notNull(clientCfg, "Destination thin client configuration.");
+
+        this.clientCfg = clientCfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void initLogger() throws Exception {
+        log = U.initLogger(null, "kafka-ignite-streamer", UUID.randomUUID(), U.defaultWorkDirectory());
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void runx() throws Exception {
+        try (IgniteClient client = Ignition.startClient(clientCfg)) {
+            this.client = client;
+
+            runAppliers();
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override protected AbstractCdcEventsApplier eventsApplier() {
+        GridBinaryMarshaller.popContext(binaryContext());
```

Comment:
Why this required?

Reply:
To deserialise `BinaryObjectImpl`. See `BinaryObjectImpl#readExternal`